### PR TITLE
Nuget + Bridge OS-specific Classes (AO-17842)

### PR DESCRIPTION
**Pull request by adamiklukasz**
https://swicloud.atlassian.net/browse/AO-17842

* v2/bindings/csharp/SnapPluginLibCs/Solarwinds.Snap.SnapPluginLib.nuspec
* removed conditional definition of dll's name; added BridgeWin and BridgeLinux classes

Validated with online solarwinds.snap.snappluginlib\1.0.5 on Windows and Linux

### CODE REVIEW

File: v2/bindings/csharp/SnapPluginLibCs/CollectorExample/CollectorExample.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
     <PropertyGroup>
         <OutputType>Exe</OutputType>
-        <TargetFramework>netcoreapp3.0</TargetFramework>
+        <TargetFramework>net5.0</TargetFramework>
```
Comment (tomekkolo) [inline v2/bindings/csharp/SnapPluginLibCs/CollectorExample/CollectorExample.csproj:5]: do we need to change that ?

Comment (adamiklukasz) [reply v2/bindings/csharp/SnapPluginLibCs/CollectorExample/CollectorExample.csproj]: yes, there is an warning during compilation that coreapp won't be supported.
Maybe net5.0 is too fresh ... I changed to netstandard2.0 (the same as lib)

Comment (tomekkolo) [reply v2/bindings/csharp/SnapPluginLibCs/CollectorExample/CollectorExample.csproj]: 👍

**Review by tomekkolo — APPROVED** [review]

**Review by tomekkolo — APPROVED** [review]

File: v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridgeLinux.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            Runner.LoadHandler loadHandler,
+            Runner.UnloadHandler unloadHandler,
+            Runner.DefineHandler defineHandler,
+            string name,
+            string version
+        );
+
+        // Collect context related functions
+
+        [DllImport(PluginLibDllName, CharSet = CharSet.Ansi, SetLastError = true)]
```
Comment (lootek) [inline v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridgeLinux.cs:27]: why not `CharSet.Unicode` ?

File: v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/SnapPluginLib.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>netstandard1.4</TargetFramework>
+        <TargetFramework>netstandard2.0</TargetFramework>
         <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
         <Copyright>Copyright (c) 2020 SolarWinds Worldwide, LLC</Copyright>
```
Comment (lootek) [inline v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/SnapPluginLib.csproj:6]: 2021

File: v2/tutorial/go.mod [hunk 4]
```diff
@@ -6,6 +6,8 @@ require (
 	github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d // indirect
 	github.com/go-ole/go-ole v1.2.4 // indirect
 	github.com/shirou/gopsutil v2.20.9+incompatible
-	github.com/solarwinds/snap-plugin-lib/v2 v2.0.0-beta-2
+	github.com/solarwinds/snap-plugin-lib/v2 v2.0.4
```
Comment (lootek) [inline v2/tutorial/go.mod:9]: why is this line here given the local replace rule?

**Review by lootek — COMMENTED** [review]
Files are missing copyright headers in general

File: v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridgeLinux.cs (hunk 2 above)
Comment (adamiklukasz) [reply v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridgeLinux.cs:27]: because the current implementation of C bridge is to build on C ANSI

File: v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/SnapPluginLib.csproj (hunk 3 above)
Comment (adamiklukasz) [reply v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/SnapPluginLib.csproj:6]: done

File: v2/tutorial/go.mod (hunk 4 above)
Comment (adamiklukasz) [reply v2/tutorial/go.mod:9]: it's for better development experience - when you are chaning lib you don't have to tag it or temporary changing branch rule in go.mod. 
Replace is ignored when module is imported directly.
Observed that this pattern is used in other repositories.

**Review by lootek — APPROVED** [review]

File: v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridge.cs [hunk 5]
```diff
@@ -1,5 +1,4 @@
 ﻿using System;
```
Comment (lootek) [inline v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridge.cs:1]: Files are missing copyright headers in general

**Review by lootek — CHANGES_REQUESTED** [review]
Files are missing copyright headers in general

File: v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridge.cs (hunk 5 above)
Comment (adamiklukasz) [reply v2/bindings/csharp/SnapPluginLibCs/SnapPluginLib/CBridge.cs]: done

**Review by tomekkolo — APPROVED** [review]

**Review by lootek — APPROVED** [review]
